base/process.jl

```diff
@@ -72,6 +72,13 @@ backticks, e.g.
 """
 Cmd
 
+hash(x::Cmd) = hash(x.exec, hash(x.env, hash(x.ignorestatus, hash(x.dir, hash(x.flags)))))
+function ==(x::Cmd, y::Cmd)
+    a = isequal(x.exec, y.exec) && isequal(x.env, y.env)
```

Comment:
Why `isequal` and not just `==`?   The floating-point semantics are irrelevant here.   Also, I don't see the need for temporary variables.  Why not just do:
```jl
==(x::Cmd, y::Cmd) = x.exec == y.exec && x.env == y.env && ....etc....
```